src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -372,11 +375,33 @@ public void cancelTask(String taskId,
         if (isRelease2EndpointsFeatureEnabled) {
             //Lock & update Task
             TaskResource task = findByIdAndObtainLock(taskId);
+            CFTTaskState previousTaskState = task.getState();
             task.setState(CFTTaskState.CANCELLED);
-            //Perform Camunda updates
-            camundaService.cancelTask(taskId);
-            //Commit transaction
-            cftTaskDatabaseService.saveTask(task);
+
+            String camundaCftTaskState = camundaService.getCftTaskState(taskId);
+
+            log.info("{} previousTaskState : {} - camundaCftTaskState : {}",
+                taskId, previousTaskState, camundaCftTaskState);
+
+            try {
+                //Perform Camunda updates
+                camundaService.cancelTask(taskId);
+                log.info("{} cancelled in camunda", taskId);
+                //Commit transaction
+                cftTaskDatabaseService.saveTask(task);
+                log.info("{} cancelled in CFT", taskId);
+            } catch (TaskCancelException ex) {
+                log.info("{} an error occurred when cancelling task. Checking Cft Task State to sync DBs", taskId);
+                if (camundaCftTaskState != null) {
+                    throw new TaskCancelException(TASK_CANCEL_UNABLE_TO_CANCEL);
```

Comment:
Are we not losing the original exception details here?

Reply:
Should add some logging for this case which includes the Camunda History cftTaskState.  Usually this would be pendingTermination and we want to know how often this occurs.

Reply:
This log is now misleading since the `isCftTaskStateExist` variable could have existed in the active table OR the history table (as per previous discussion point).

Reply:
will remove this one